spec/support/shared_validator_specs.rb

```diff
@@ -4,12 +4,24 @@ def entity_stub(entity, property, val)
   instance
 end
 
+def entity_stub_data(entity, data={})
+  OpenStruct.new(FactoryGirl.attributes_for(entity).merge(data))
```

Comment:
you can replace the `entity_stub` method above with this and call it with `entity_stub(entity, property => val)` from existing specs.